src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ClusterApplierServiceStatsCollector.java

```diff
@@ -0,0 +1,164 @@
+                return;
+            }
+            mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+            mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+            ClusterApplierServiceStats clusterApplierServiceStats = null;
+            try {
+                clusterApplierServiceStats = mapper.readValue(
+                        mapper.writeValueAsString(getClusterApplierServiceStats()), ClusterApplierServiceStats.class);
+            } catch (InvocationTargetException | IllegalAccessException | NoSuchMethodException ex) {
+                LOG.warn("No method found to get cluster state applier thread stats. " +
+                        "Skipping ClusterApplierServiceStatsCollector");
+                return;
+            }
+
+            ClusterApplierServiceMetrics clusterApplierServiceMetrics = new ClusterApplierServiceMetrics(
+                    computeLatency(clusterApplierServiceStats), computeFailure(clusterApplierServiceStats));
+
+            value.setLength(0);
+            value.append(PerformanceAnalyzerMetrics.getJsonCurrentMilliSeconds())
+                    .append(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor);
+            value.append(clusterApplierServiceMetrics.serialize());
+            saveMetricValues(value.toString(), startTime);
+            this.tracker = new MetricsTracker(clusterApplierServiceStats.timeTakenInMillis,
+                    clusterApplierServiceStats.failedCount, clusterApplierServiceStats.totalCount);
+            PerformanceAnalyzerApp.WRITER_METRICS_AGGREGATOR.updateStat(
+                    WriterMetrics.CLUSTER_APPLIER_SERVICE_STATS_COLLECTOR_EXECUTION_TIME, "",
+                    System.currentTimeMillis() - mCurrT);
+        } catch (Exception ex) {
+            PerformanceAnalyzerApp.ERRORS_AND_EXCEPTIONS_AGGREGATOR.updateStat(
+                    ExceptionsAndErrors.CLUSTER_APPLIER_SERVICE_STATS_COLLECTOR_ERROR, "", 1);
+            LOG.debug("Exception in Collecting Cluster Applier Service Metrics: {} for startTime {}",
+                    () -> ex.toString(), () -> startTime);
+        }
+    }
+
+    private Object getClusterApplierServiceStats() throws InvocationTargetException, IllegalAccessException,
+            NoSuchMethodException {
+        Method method = ClusterApplierService.class.getMethod(GET_CLUSTER_APPLIER_SERVICE_STATS_METHOD_NAME);
+        return method.invoke(ESResources.INSTANCE.getClusterService().getClusterApplierService());
+    }
+
+    private long computeLatency(final ClusterApplierServiceStats currentMetrics) {
+        final long rate = computeRate(tracker.getPrevTotalCount());
+        if(rate == 0) {
+            return 0L;
+        }
+        return (currentMetrics.timeTakenInMillis - tracker.getPrevTimeTakenInMillis()) / rate;
+    }
+
+    private long computeRate(final long currentTotalCount) {
```

Comment:
All these methods would return 0 in case there is no update to cluster state since last retrieval. Just that its handling should take care of it accordingly. Javadocs here would help providing the context.

Reply:
Added javadoc